BaseApi.js

```diff
@@ -68,8 +68,14 @@ const makeGetRequest = (url, token = null) => {
     method: 'GET',
     headers: getHeaders(token)
   })
-    .then(response => response.json())
-    .then(unpackedData => unpackedData)
+    .then(response => {
+      const statusCode = response.status
+      const data = response.json()
+      return Promise.all([ statusCode, data ])
```

Comment:
does it work the same way without Promise.all() here? i think statusCode is not a Promise, but only data.

Reply:
maybe more clear to return `{ statusCode, data }` and in the next `then` also return `{ statusCode, data}` instead of `values[0], values[1]`?

Reply:
I copied this from the post method in BaseApi, so if we change this we should probably change it there as well.
Apart from that, as data is a Promise we do need to return a Promise, otherwise the subsequent `then` calls fail, right?

Reply:
could be yes, i was just wondering if it would resolve, but probably if we have at least one promise the array we need it. sorry, you can ignore this bit @Rineee

Reply:
I think it's worth testing, as I wondered about it too! :)

Reply:
I did test that at the time I wrote it for the post method and it does not work without the promise..